hall-of-fame-service/src/main/java/hcmus/alumni/halloffame/controller/HallOfFameServiceController.java

```diff
@@ -0,0 +1,243 @@
+		if (!emailOfUser.equals("")) {
+			userId = halloffameRepository.getUserIdByEmail(emailOfUser);
+		}
+		else {
+			return ResponseEntity.status(HttpStatus.NOT_FOUND).body("Email not existed");
+		}
+		UserModel userModel = null;
+		if (!userId.equals("")) {
+			userModel = new UserModel(userId);
+		}
+		else {
+			return ResponseEntity.status(HttpStatus.NOT_FOUND).body("User not existed");
+		}
+		try {
+			// Save thumbnail image
+			String thumbnailUrl = imageUtils.saveImageToStorage(imageUtils.getHofPath(id), thumbnail, "thumbnail");
+			// Save hall of fame to database
+			HallOfFameModel halloffame = new HallOfFameModel(id, new UserModel(creator), title, thumbnailUrl,
+					new FacultyModel(facultyId), userModel, beginningYear);
+			// Lên lịch
+			if (scheduledTimeMili != null) {
+				halloffame.setPublishedAt(new Date(scheduledTimeMili));
+				halloffame.setStatus(new StatusPostModel(1));
+			} else {
+				halloffame.setPublishedAt(new Date());
+			}
+			if (!facultyId.equals(0)) {
+				halloffame.setFaculty(new FacultyModel(facultyId));
+			}
+			halloffameRepository.save(halloffame);
+		} catch (IOException e) {
+			System.err.println(e);
+			return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body("Failed to save thumbnail");
+		}
+		return ResponseEntity.status(HttpStatus.CREATED).body(id);
+	}
+
+	@PreAuthorize("hasAnyAuthority('Admin')")
+	@PutMapping("/{id}")
+	public ResponseEntity<String> updateHallOfFame(@PathVariable String id,
+			@RequestParam(value = "title", defaultValue = "") String title,
+			@RequestParam(value = "thumbnail", required = false) MultipartFile thumbnail,
+			@RequestParam(value = "faculty", defaultValue = "") Integer faculty,
+			@RequestParam(value = "emailOfUser") String emailOfUser,
+			@RequestParam(value = "beginningYear", required = false) Integer beginningYear) {
+
+		try {
+			// Find hall of fame
+			Optional<HallOfFameModel> optionalHallOfFame = halloffameRepository.findById(id);
+			if (optionalHallOfFame.isEmpty()) {
+				return ResponseEntity.status(HttpStatus.NOT_FOUND).body("Invalid id");
+			}
+			HallOfFameModel halloffame = optionalHallOfFame.get();
+
+			// Update fields if provided
+			if (!title.equals("")) {
+				halloffame.setTitle(title);
+			}
+			if (thumbnail != null && !thumbnail.isEmpty()) {
+				// Check if thumbnail size exceeds the limit
+				if (thumbnail.getSize() > 5 * 1024 * 1024) {
+					return ResponseEntity.status(HttpStatus.BAD_REQUEST).body("File must be lower than 5MB");
+				}
+
+				// Save new thumbnail
+				String thumbnailUrl = imageUtils.saveImageToStorage(imageUtils.getHofPath(id), thumbnail, "thumbnail");
+				halloffame.setThumbnail(thumbnailUrl);
+			}
+			if (!faculty.equals("")) {
+				halloffame.setFaculty(new FacultyModel(faculty));
+			}
+			if (beginningYear != null) {
+				halloffame.setBeginningYear(beginningYear);
+			}
+
+			if (!emailOfUser.equals("")) {
+				String userId = halloffameRepository.getUserIdByEmail(emailOfUser);
```

Comment:
```suggestion
				String userId = halloffameRepository.getUserIdByEmail(emailOfUser);
				if (userId.isEmpty()) {
					return ResponseEntity.status(HttpStatus.NOT_FOUND).body("User not existed");
				}
```
thấy tim mà ko xử lý nếu ko tìm dc userid v :(